scripts/clear-package-dir.js

```diff
@@ -0,0 +1,137 @@
+let argv = yargs
+  .option(`dry-run`, {
+    default: false,
+    describe: `Don't delete files - just show what would be deleted`,
+  })
+  .option(`verbose`, {
+    default: false,
+    describe: `Show files that would be bundled and mark files that will be deleted`,
+  })
+  .option(`yes`, {
+    default: false,
+    describe: `Force deletion of file without prompting user to confirm`,
+  }).argv
+
+const _verbose = argv[`dry-run`] || argv[`verbose`] || !argv[`yes`]
+
+const buildIgnoreArray = str =>
+  str
+    .split(`\n`)
+    .filter(line => {
+      // skip empty lines and comments
+      if (!line || line[0] === `#`) {
+        return false
+      }
+
+      return true
+    })
+    .reduce((acc, item) => {
+      acc.push(item)
+
+      // add "<directory>/**" glob as ignore package need that to
+      // properly ignore entries like "node_modules"
+      if (!/\*\*$/.test(item)) {
+        acc.push(`${item}/**`)
+      }
+      return acc
+    }, [])
+
+const getListOfFilesToClear = ({ location, name }) => {
+  if (_verbose) {
+    console.log(`Files that will be packed for ${chalk.bold(name)}:`)
+  }
+
+  let gitignore = []
+  try {
+    gitignore = buildIgnoreArray(
+      fs.readFileSync(path.join(location, `.gitignore`), `utf-8`)
+    )
+  } catch {
+    // not all packages have .gitignore - see gatsby-plugin-no-sourcemap
+  } finally {
+    const notTrackedFiles = execSync(
+      `git ls-files --others --exclude-standard`,
+      {
+        cwd: location,
+      }
+    )
+      .toString()
+      .split(`\n`)
+
+    gitignore = gitignore.concat(notTrackedFiles)
+
+    // we need main file - for now hardcode index
+    // that is used in most of our packages
+    // it should actually be in each package .gitignore
+    gitignore.push(`!/index.js`)
+  }
+
+  let npmignore = []
+  try {
+    npmignore = buildIgnoreArray(
+      fs.readFileSync(path.join(location, `.npmignore`), `utf-8`)
+    )
+  } catch {
+    // not all packages have .npmignore - see gatsby-plugin-no-sourcemap
+  } finally {
+    npmignore = npmignore.concat([`node_modules/**`])
+  }
+
+  const result = glob.sync(`**/*`, {
+    nodir: true,
+    ignore: npmignore,
+    cwd: location,
+  })
+
+  const ig = ignore().add(gitignore)
+
+  const filesToDelete = result
+    .filter(file => {
+      const willBeDeleted = ig.ignores(file)
+      if (_verbose) {
+        console.log(
+          `[ ${willBeDeleted ? chalk.red(`DEL`) : chalk.green(` - `)} ] ${file}`
+        )
+      }
+
+      return willBeDeleted
+    })
+    .map(file => path.join(location, file))
+
+  return filesToDelete
+}
+
+try {
+  const changed = JSON.parse(
+    execSync(
+      `node_modules/.bin/lerna changed --json --loglevel=silent`
+    ).toString()
+  )
+  const filesToDelete = _.flatten(changed.map(getListOfFilesToClear))
+
+  if (!argv[`dry-run`] && filesToDelete.length > 0) {
+    if (
+      argv[`force`] ||
+      PromptUtilities.confirm(`Are you sure you want to delete those files?`)
```

Comment:
This returns a Promise! (ran into this locally, doesn't actually wait for me to prompt and deletes the files).

Annoyingly, we could make this whole thing async--not sure if there's a better solution, e.g.

```js
async function clearPackgeDir() {
 // all this logic, with the one await block
}

clearPackageDir()
```

Reply:
ops, yeah, will wrap entire thing in async function, seems like easiest solution here